tools/detect_nonexistent_testdata

```diff
@@ -1,14 +1,55 @@
-#!/bin/sh -e
-
-FILES="${@}"
-success=1
-for FILE in $FILES; do
-	if  ! [ -f "$FILE" ]; then
-		echo "Error! $FILE does not exist."
-		success=0
-	elif ! [ -s "$FILE" ]; then
-		echo "Warning! The test_data file $FILE is empty."
-	fi
-done
-[ $success = 1 ] && echo "SUCCESS" && exit 0
-exit 1
+#!/usr/bin/env perl
+
+# The script verifies if test data files included in yaml schedule exist in the repo.
+
+use strict;
+use warnings;
+use YAML::Tiny;
+use File::Basename;
+use Data::Dumper;
+
+=head2 get_testdata_filename
+
+  get_testdata_paths($schedule_file_path);
+
+Returns yaml test data file name(s) with path, in case yaml schedule includes any, either via $include tag or YAML_TEST_DATA variable.
+
+=cut
+sub get_data_files {
+    my ($schedule_file_path) = @_;
+    my $schedule             = YAML::Tiny::LoadFile($schedule_file_path);
+    my $include_tag          = '$include';
+    my @filename_list;
+    if ($schedule->{test_data}) {
+        my $test_data = $schedule->{test_data};
+        my @filenames;
+        if (ref($test_data->{$include_tag}) eq 'ARRAY') {
+            @filenames = @{$test_data->{$include_tag}};
+        }
+        else {
+            @filenames = $test_data->{$include_tag};
```

Comment:
Ok, now I see the problem, it's in this line. If include tag is `undef` you add it to the @filenames.
Try
```perl
my @arr = undef;
print "lala" if(@arr);
```